org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/contentmodel/XMLSchemaHoverExtensionsTest.java

```diff
@@ -305,6 +315,80 @@ public void hoverTextWithUnion3() throws BadLocationException, MalformedURIExcep
 				r(2, 52, 3, 9));
 	}
 
+	@Test
+	public void hoverWithNullParentNode() throws Exception {
+		ContentModelHoverParticipant hoverParticipant = new ContentModelHoverParticipant();
+		IHoverRequest hoverRequest = new IHoverRequest() {

```

Comment:
We should prolly start using a mocking library like [Mockito](https://site.mockito.org/)